decred/decred/dcr/rpc.py

```diff
@@ -41,11 +47,65 @@ def stringify(thing):
     return thing
 
 
+class Response:
+    """Parses a JSON-RPC 2.0 response."""
+
+    def __init__(self, jsonRep):
+        """
+        Args:
+            jsonRep (str or dict): The raw result from the server.
+        """
+        msg = json.loads(jsonRep) if isinstance(jsonRep, str) else jsonRep
+        self.id = msg.get("id")
+        self.result = msg.get("result")
+        self.error = msg.get("error")
+
+
+class Request:
+    """A JSON-RPC 2.0 request."""
+
+    def __init__(self, reqID, method, params):
+        """
+        Args:
+            reqID (int): The request ID.
+            method (str): The JSON-RPC method.
+            params (list): JSON-serializable parameters.
+        """
+        self.id = reqID
+        self.method = method
+        self.params = params
+
+    def dict(self):
+        """
+        Dump the Request as a Python dictionary.
+
+        Returns:
+            dict: The Request encoded as a Python dictionary.
+        """
+        return {
+            "jsonrpc": "2.0",
+            "id": self.id,
+            "method": self.method,
+            "params": self.params,
+        }
+
+    def json(self):
+        """
+        Dump the request as a JSON-formatted string.
+
+        Returns:
+            str: The Request encoded as a JSON string.
+        """
+        return json.dumps(self.dict())
+
+
 class Client:
     """
     The Client communicates with the blockchain RPC API.
     """
 
+    reqID = 0
```

Comment:
Please name this `nextAvailableReqID` (as opposed to `lastUsedReqID`) (or add a comment to that effect, but a more explicit name is preferable).